core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -87,25 +107,53 @@ public void install(Channel channel) {
 
         // set up ping from both directions, so that in case of a router dropping a connection,
         // both sides can notice it and take compensation actions.
-        setUpPingForChannel(channel, pingInterval);
+        setUpPingForChannel(channel, pingTimeoutSeconds, pingIntervalSeconds);
     }
 
-    private static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
-        private static final long serialVersionUID = -2702219700841759872L;
-        private int pingInterval;
-        public SetUpRemotePing(int pingInterval) {
-            this.pingInterval = pingInterval;
+    static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
+        private final int pingTimeoutSeconds;
+        private final int pingIntervalSeconds;
+
+        SetUpRemotePing(int pingTimeoutSeconds, int pingIntervalSeconds) {
+            this.pingTimeoutSeconds = pingTimeoutSeconds;
+            this.pingIntervalSeconds = pingIntervalSeconds;
         }
 
+        @Override
         public Void call() throws IOException {
-            setUpPingForChannel(Channel.current(), pingInterval);
+            setUpPingForChannel(Channel.current(), pingTimeoutSeconds, pingIntervalSeconds);
             return null;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            SetUpRemotePing that = (SetUpRemotePing) o;
+            return this.pingTimeoutSeconds == that.pingTimeoutSeconds
+                && this.pingIntervalSeconds == that.pingIntervalSeconds;
+        }
+
+        @Override
+        public int hashCode() {
+            // TODO(deadmoose): switch to Objects.hash once Java 7's fully available
```

Comment:
I would have thought a simple arithmetic expression with a prime would work here (such as is auto generated by eclipse)